src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```diff
@@ -246,5 +279,15 @@ internal void PerformOperation(DacFxOperation operation)
         {
             operation.Execute(TaskExecutionMode.Execute);
         }
+
+        /// <summary>
+        /// For testing purpose only
+        /// </summary>
+        /// <param name="operation"></param>
+        internal string PerformGenerateDeployPlanOperation(GenerateDeployPlanOperation operation)
```

Comment:
Do we need this method? It seems so similiar to PerformOperation() above. Can PerformOperation just return a string?

Reply:
it isn't needed anymore after adding the deploy report property to the operation